packages/core/src/internal/clients/cache/AbstractCacheClient.ts

```diff
@@ -65,26 +65,39 @@ import {
 } from './ICacheClient';
 import {IControlClient} from './IControlClient';
 import {IDataClient} from './IDataClient';
+import {IPingClient} from './IPingClient';
 
 export abstract class AbstractCacheClient implements ICacheClient {
   // making these protected until we fully abstract away the nodejs client
   protected readonly controlClient: IControlClient;
   protected readonly dataClients: IDataClient[];
+  // TODO: Make pingClient required if and when the nodejs side starts adding
+  //  one as well
```

Comment:
can you just go ahead and add it to the node.js sdk while you are in here?  but I do think you still want this variable to be `?` for the reasons I mentioned above.  And then this variable becomes private, and you provide a protected accessor function that lazily instantiates the client when it is called for the first time.

Reply:
Nice, makes sense, will do.